torch/_subclasses/fake_tensor.py

```diff
@@ -1441,6 +1453,14 @@ def _validate_cache_key(
         # Avoid caching for any ops that would require a more sophisticated
         # caching implementation, e.g., data dependent ops or ops that modify
         # the inputs.
+        from torch._higher_order_ops.utils import registered_hop_fake_fns
+
```

Comment:
type of `func` is now wrong, need to update it to be a union

Reply:
So, I tried this earlier and this causes  a bunch of mypy failures because now func is expecting `tags` and other attributes all over the place, which are not present in HigherOrderOperator.

Reply:
yeah don't worry about it